lib/matrix_client/session.ex

```diff
@@ -120,8 +154,40 @@ defmodule MatrixClient.Session do
 
   def filter_invite_events(events) do
     Enum.filter(events, fn event ->
-      %{"content" => content} = event
-      content == %{"join_rule" => "invite"}
+      %{"type" => type} = event
+      type == "m.room.join_rules"
+    end)
+  end
+
+  def sync_leave({room_id, _}, rooms) do
+    Map.delete(rooms, room_id)
+  end
+
+  def sync_alias({room_id, room_data}, aliases) do
+    %{"timeline" => %{"events" => events}} = room_data
+
+    case filter_alias_event(events) do
+      [ae] -> alias_helper_ids(aliases, room_id, ae["content"]["alias"])
+      _ -> aliases
+    end
+  end
+
+  defp alias_helper_ids(aliases, room_id, a) do
```

Comment:
Not sure what these functions are doing just by reading the name? In general I'm not a big fan of using `helper` as a descriptor as it's so vague. Thoughts?